test/unit/platforms/detect/os_common_test.rb

```diff
@@ -16,26 +16,14 @@ def initialize
 
   describe 'winrm? check' do
     it 'return winrm? true' do
-      require 'train/transports/winrm'
-      be = Train::Transports::WinRM::Connection.new(nil)
-      detector.instance_variable_set(:@backend, be)
+      OsDetectLinuxTester.any_instance.stubs(:backend_name).returns('TrainPlugins::WinRM::Connection')
       detector.winrm?.must_equal(true)
     end
 
-    it 'return winrm? false when winrm is loaded' do
-      require 'train/transports/winrm'
-      be = mock('Backend')
-      detector.instance_variable_set(:@backend, be)
-      detector.winrm?.must_equal(false)
-    end
-
     it 'return winrm? false when winrm is not loaded' do
-      require 'train/transports/winrm'
-      winrm = Train::Transports.send(:remove_const, 'WinRM')
       be = mock('Backend')
       detector.instance_variable_set(:@backend, be)
       detector.winrm?.must_equal(false)
```

Comment:
Why not set up this test similar to the `true` case in order to make it obvious that it is just the backend_name that's really causing `winrm?` to return true vs false? Symmetry is king in these sorts of tests.

Reply:
Fine